traffic_ops/traffic_ops_golang/deliveryservice/health.go

```diff
@@ -132,7 +160,24 @@ func addHealth(ds tc.DeliveryServiceName, data map[tc.CacheGroupName]tc.HealthDa
 		if cache.CacheGroup == nil {
 			continue // TODO warn?
 		}
-
+		if topology != "" {
+			if _, ok := cacheGroupNameMap[*cache.CacheGroup]; !ok {
+				continue
+			}
+			cacheCapabilities = make(map[string]bool)
```

Comment:
This shadows a variable of the same name in an outer scope; it should be renamed to avoid collisions and confusion.

Also: nit but you know how big this needs to be. It's exactly the size of the cache server's `Capabilities` array, since it's not legal for those to contain duplicates. So you can construct with that in mind to cut down on re-allocations.

Also the value of each map key is unused, so to save an insignificant amount of space this could just be a `map[string]struct{}`